UnitySetup/UnitySetup.psm1

```diff
@@ -125,8 +125,18 @@ class UnityProjectInstance {
         $projectSettingsFile = [io.path]::Combine($path, "ProjectSettings\ProjectSettings.asset")
         if (!(Test-Path $projectSettingsFile)) { throw "Project is missing ProjectSettings.asset" }
 
-        $prodName = ((Get-Content $projectSettingsFile -Raw | ConvertFrom-Yaml)['playerSettings'])['productName']
-        if (!$prodName) { throw "ProjectSettings is missing productName" }
+        try { 
+            $prodName = ((Get-Content $projectSettingsFile -Raw | ConvertFrom-Yaml)['playerSettings'])['productName']
+            if (!$prodName) { throw "ProjectSettings is missing productName" }
+        }
+        catch {
+            Write-Warning -Message "An Exception was caught!"
+            Write-Warning -Message "Exception Type: $($_.Exception.GetType().FullName)"
+            Write-Warning -Message "Exception Message: $($_.Exception.Message)"
+        }
+        finally{
+            $prodName = ""
```

Comment:
by doing this in `finally`, `$prodName` will always end up as the empty string. I'd move this  line up into the end of the `catch` block and probably set it to `$null` instead of the empty string.

Reply:
fixed, thanks :)